src/System.Runtime.Numerics/src/System/Numerics/BigCalc.SquMul.cs

```diff
@@ -0,0 +1,356 @@
+﻿using System.Diagnostics;
+using System.Security;
+
+namespace System.Numerics
+{
+    internal static partial class BigCalc
+    {
+#if DEBUG
+
+        private const int SquareThreshold = 128 / 32;
```

Comment:
I would be very helpful if you could add some comments describing how you arrived at these thresholds and what they are used for. And more generally, it would be great to see more comments in this file. Understanding the code requires some domain specific background, and it would be great if comments pointed future code maintainers to such background. For a great example, see https://github.com/dotnet/corefx/blob/master/src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs implemented and commented by @GrabYourPitchforks who points to lots of good background documents explaining various Unicode concepts.